core/src/main/kotlin/org/evomaster/core/problem/api/service/ApiWsStructureMutator.kt

```diff
@@ -208,10 +208,10 @@ abstract class ApiWsStructureMutator : StructureMutator() {
                 addedInsertions.add(a)
             }
 
-            // Removing the existing action added with the default WireMock address
-            val defaultActions = old.filter { it.hostname == a.hostname && it.localIPAddress == ExternalServiceSharedUtils.DEFAULT_WM_LOCAL_IP };
-            if (defaultActions.isNotEmpty()) {
-                ind.removeHostnameResolutionAction(defaultActions)
+            // Removing the existing default IP address action
+            val defaultHNAction = old.filter { it.hostname == a.hostname && it.localIPAddress != a.localIPAddress }
```

Comment:
this is doing something different from what stated in the comment. you delete any other previous mapping for that hostname having a different IP, regardless of whether it is default or not. either code should be changed, or the comment

Reply:
This is fixed under https://github.com/EMResearch/EvoMaster/pull/958